vllm/model_executor/models/baichuan.py

```diff
@@ -303,16 +303,18 @@ def forward(
     def load_weights(self,
                      model_name_or_path: str,
                      cache_dir: Optional[str] = None,
-                     use_np_cache: bool = False):
+                     load_format: str = "auto"):
         tp_world_size = get_tensor_model_parallel_world_size()
         tp_rank = get_tensor_model_parallel_rank()
         state_dict = self.state_dict()
 
         for name, loaded_weight in hf_model_weights_iterator(
-                model_name_or_path, cache_dir, use_np_cache):
+                model_name_or_path, cache_dir, load_format):
             if "rotary_emb.inv_freq" in name:
                 continue
 
+            loaded_weight = convert_pyslice_to_tensor(loaded_weight)
```

Comment:
Could you explain why `convert_pyslice_to_tensor` is applied selectively? It seems the function is never called for some models. And for some other models, it is only called for QKV weights.

Reply:
See below for detailed comments.